File: Examples/PY/test20.py
def insert[A](l: list[A], index: int, elt: A) -> list[A]:
    if (index < 0) or (index >= (len(l))):
        raise IndexError
    else:
        return [*l[0:index], elt, *l[index:]]
def update[A](l: list[A], index: int, elt: A) -> list[A]:
    if (index < 0) or (index > (len(l))):
        raise IndexError
    else:
        return [*l[0:index], elt, *l[index + 1:]]
l = [1, 2, 3, 4, 5]

print(insert(l, 3, 9))

File: Examples/TS/test20.ts
function insert<A>(l: A[], index: number, elt: A): A[] {
    if ((index < 0) || (index >= len(l))) {
        throw new RangeError()
    } else {
        return [...l.slice(0, index), elt, ...l.slice(index)]
    }
}

function update<A>(l: A[], index: number, elt: A): A[] {
    if ((index < 0) || (index > len(l))) {
        throw new RangeError()
    } else {
        return [...l.slice(0, index), elt, ...l.slice(index + 1)]
    }
}
const l = [1, 2, 3, 4, 5]

console.log(insert(l, 3, 9))
